### ISSUE TITLE
Document Sobelow.XSS.SendResp

### ISSUE THREAD

**Issue by jared-mackey** [nccgroup/sobelow#69]
Please provide documentation on the `Sobelow.XSS.SendResp` lint including ideal resolution.

**Comment by sigwinch28** [comment]
Yeah, I've just has this come up as a "medium confidence" using `Plug.Conn`'s `send_resp` even though no user-provided data is being included in the response.

**Comment by devstopfix** [comment]
I have it - I am passing the POST request into a function, but returning a hard-coded string

```elixir
defp handle_body(conn, body) do
  case parse(body) do
    {:ok, tree} ->
      Conn.assign(conn, :tree, tree)

    {:error, _fatal} ->
      conn
      |> send_resp(@bad_request, invalid_xml_body())
      |> halt
  end
end

@spec invalid_xml_body :: String.t()
def invalid_xml_body, do: "INVALID XML"
```

The complaint:

```json
"medium_confidence": [
  {
     "file": "lib/XXXX/plugs/xml.ex",
     "line": 55,
     "type": "XSS.SendResp: XSS in `send_resp`",
     "variable": "invalid_xml_body"
  }
]
},
"sobelow_version": "0.10.4",
"total_findings": 1
```

**Comment by atonse** [comment]
We're still encountering this problem and it doesn't seem terribly useful without any documentation.

We are using a content engine to generate the string for the entire HTML page, and then using conn |> send_resp(:ok, content) with the returned content.

It's in a `GET` request too.

**Comment by asad-nl** [comment]
Any updates? Running into the same issue.

**Comment by cghtompkins17** [comment]
I had a same issue where I was returning an API response from the controller. The controller action was this : 
```elixir
defmodule CauldronWeb.ApiController do
  use CauldronWeb, :controller

  # Some processing library
  alias Cauldron.RequestProcessor

  def api_specs(conn, params) do
    {status_code, response_body} = RequestProcessor.handle_get_api_spec_request(params)

    conn
    |> send_resp(status_code, response_body |> Jason.encode!())
  end
end
```

When I ran the mix sobelow command, I got the warning 
```JSON
{
  "findings": {
    "high_confidence": [],
    "low_confidence": [],
    "medium_confidence": [
      {
        "file": "lib/cauldron_web/controllers/api_controller.ex",
        "line": 11,
        "type": "XSS.SendResp: XSS in `send_resp`",
        "variable": "|>"
      }
    ]
  },
  "sobelow_version": "0.11.1"
}
```

To fix this I added the `put_resp_content_type("application/json")` before sending response. (I am having a JSON API)

The final code looked something like this
```elixir
defmodule CauldronWeb.ApiController do
  use CauldronWeb, :controller

  alias Cauldron.RequestProcessor

  def api_specs(conn, params) do
    {status_code, response_body} = RequestProcessor.handle_get_api_spec_request(params)

    conn
    |> put_resp_content_type("application/json")
    |> send_resp(status_code, response_body |> Jason.encode!())
  end
end
```

**Comment by justincy** [comment]
Sobelow can't always know whether any given variable is user supplied input so it appears to be assuming that it could be if the response body is generated dynamically in any way, e.g. via a function call. That would be why @devstopfix was able to [fix one problem](https://github.com/nccgroup/sobelow/issues/69#issuecomment-767112765) by moving the body from a method to a module attribute.

At least I think that's how it works. [The Sobelow code](https://github.com/nccgroup/sobelow/blob/master/lib/sobelow/xss/send_resp.ex) is pretty hard to follow. I can't tell how this would ever generate a medium confidence issue but that's what it's doing.
